### PR TITLE
Fix informer-gen to honor nonNamespaced tag

### LINKED ISSUE #80457: Could not compile generated informer with +genclient:nonNamespaced
**What happened**:

if I annotate non namespaced CRD type like this:

```go
// ...
// +genclient                                                                                                                                                                     
// +genclient:nonNamespaced                                                                                                                                                       
// ...
type Foo struct {
...
```

informer-gen generates code which cannot compile:

```
unknown field 'namespace' in struct literal of type fooInformer
```

It looks like nonNamspaced tag is not respected somewhere.

**What you expected to happen**:

Generated code should compile just fine.

**How to reproduce it (as minimally and precisely as possible)**:

See what happened.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.14
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by tatsuhiro-t**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change

/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

This PR fixes #80457 so that informer-gen produces correct code for a type with nonNamespaced tag.

**Which issue(s) this PR fixes**:

Fixes #80457

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Fixes the bug in informer-gen that it produces incorrect code if a type has nonNamespaced tag set.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by yliaog** [comment]
please add a test case for this fix

**Comment by tatsuhiro-t** [comment]
Where is the existing test case?

**Comment by yliaog** [comment]
it looks there is not one currently. but still it is good to add one.

**Comment by tatsuhiro-t** [comment]
I have no time to add test case **from scratch**.   I think it is much better and beneficial to everyone to merge this simple fix rather than leaving code-generator in miserable, unusable state.

**Comment by yliaog** [comment]
please see my comment in https://github.com/kubernetes/kubernetes/issues/80457, i think the current informer gen is working as intended.

**Comment by tatsuhiro-t** [comment]
I disagree. The other generators can handle this without extra blank lines. Actually, informer-gen partially handles this except for the broken portion which #80458 tries to fix.
Leaving generator tool like this in inconsistent state is huge UX regression. This is clearly a bug to me.

**Comment by yliaog** [comment]
ok, i agree with you. could you maybe modify the test in the following directories to make sure it won't regress.

staging/src/k8s.io/code-generator/hack/verify-codegen.sh
staging/src/k8s.io/code-generator/_examples/apiserver/informers/

**Comment by tatsuhiro-t** [comment]
904dc3e62e0628fa49249779687cdf730c9ac4c0 modifies test to exercise nonNamespaced tag

File: staging/src/k8s.io/code-generator/_examples/apiserver/apis/example2/types.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
 // TestType is a top-level type. A client is created for it.
+// +genclient:nonNamespaced
 type TestType struct {
```
Comment (yliaog) [inline staging/src/k8s.io/code-generator/_examples/apiserver/apis/example2/types.go]: nit: maybe remove the blank line #23, and switch the added line 25 with 24?

Comment (tatsuhiro-t) [reply staging/src/k8s.io/code-generator/_examples/apiserver/apis/example2/types.go]: Done.

**Comment by trierra** [comment]
what else is required for this PR to be merged? Maybe I can help?

**Comment by yliaog** [comment]
it needs approval.
@sttts could you please take a look?

**Comment by tatsuhiro-t** [comment]
Is there any blocking issue to approve this PR?

**Comment by tatsuhiro-t** [comment]
@yliaog @sttts 1 month passed since I got LGTM from @yliaog 
Any chance to approve this PR?

**Comment by tatsuhiro-t** [comment]
@sttts @lavalamp @yliaog  ping

**Comment by nikhita** [comment]
super tiny nit: can you squash the last two commits together?

/assign @sttts 
for approval

**Comment by tatsuhiro-t** [comment]
@nikhita Squashed 2 last commits into one and force-pushed.  Thank you.

**Review by nikhita — APPROVED** [review]
/lgtm

**Comment by nikhita** [comment]
node-e2e is currently broken: https://github.com/kubernetes/kubernetes/issues/83086

**Pull request merged**